source/code/troubleshooter/modules/connect/check_endpts.py

```diff
@@ -0,0 +1,83 @@
+# INSPIRED BY update_mgmt_health_check.py
+
+import socket
+
+from error_codes import *
+from errors      import error_info
+from helpers     import geninfo_lookup
+
+OMSADMIN_PATH = "/etc/opt/microsoft/omsagent/conf/omsadmin.conf"
+
+# socket connect to specific endpoint
+def check_endpt(endpoint):
+    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    try:
+        response = sock.connect_ex((endpoint, 443))
+        return (response == 0)
+    except Exception:
+        return False
+
+
+
+# check general internet connectivity
+def check_internet_connect():
+    if (check_endpt("bing.com") and check_endpt("google.com")):
```

Comment:
Is there some generic network connection check we can use that is supposed to be more-less universal across linux systems, e.g. some update service URL, world clock or similar? I feel like bing.com/google.com can be blocked on purpose while internet in general is still available, but I do not know if there is better thing to use...

Reply:
I haven't been able to find one is the issue, since technically customers only need to whitelist the endpoints given [on the Azure Docs page](https://docs.microsoft.com/en-us/azure/azure-monitor/platform/log-analytics-agent#firewall-requirements)